bot/cogs/snakes.py

```diff
@@ -29,6 +34,25 @@ def __init__(self, bot: AutoShardedBot):
         :return: A dict containing information on a snake
         """
 
+        # Getting the summary
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get('https://en.wikipedia.org/w/api.php?action=parse&format=json&page=' + name + '&section=1') as resp:
+                text_data = await resp.json()
+                text = json.loads(text_data)
+                html = text["text"]["*"]
+                soup = BeautifulSoup(html)
+                summary = soup.find('p')
+
+        # Getting the image
+
+        async with aiohttp.ClientSession() as session:
```

Comment:
You could remove this `with` block and just use the one you already have above.

Reply:
Meaning put everything under 1 session?